tests/integration/components/new-signup/button-test.js

```diff
@@ -42,31 +57,46 @@ module('Integration | Component | sign-up/button', function (hooks) {
   test('Button should be of type button', async function (assert) {
     assert.expect(1);
     this.set('inputValue', 'Get Started');
+    this.setProperties({
+      onClick: function () {
+        this.currentStep = this.SECOND_STEP;
+      },
+    });
 
     await render(hbs`
-      <SignUp::Button>{{inputValue}}</SignUp::Button>
+      <Button @onClick={{this.onClick}}>{{inputValue}}</Button>
     `);
 
     assert.dom('[data-test-signup-button]').hasAttribute('type', 'button');
   });
 
-  test('Button should have spinner if @isSubmitClick is true', async function (assert) {
+  test('Button should have spinner if @isLoading is true', async function (assert) {
     assert.expect(1);
     this.set('inputValue', 'Get Started');
+    this.setProperties({
+      onClick: function () {
+        this.currentStep = this.SECOND_STEP;
+      },
+    });
 
     await render(hbs`
-      <SignUp::Button @isSubmitClicked={{true}}>{{inputValue}}</SignUp::Button>
+      <Button @onClick={{this.onClick}} @isLoading={{true}}>{{inputValue}}</Button>
     `);
 
     assert.dom('[data-test-signup-button-spinner]').exists();
   });
 
-  test('Button should not have spinner if @isSubmitClick is not true', async function (assert) {
+  test('Button should not have spinner if @isLoading is not true', async function (assert) {
```

Comment:
Nit: we should add tests to check onClick

Reply:
yes we can do it in another PR